### PR TITLE
Added a function to load the directories in the optional file: extra_model_paths.yaml

### PR DESCRIPTION
**Pull request by dimtoneff**
- Added a function to load the directories in the optional file: extra_model_paths.yaml
- Added some files & dirs to .gitignore
- Made the script work on my embedded_python on Win10 environment

### CODE REVIEW

**Comment by dimtoneff** [comment]
> extra_model_paths.yaml found: J:\SD\ComfyUI\ComfyUI\extra_model_paths.yaml
> Adding extra search path checkpoints J:\SD\stable-diffusion-webui\models/Stable-diffusion
> Adding extra search path configs J:\SD\stable-diffusion-webui\models/Stable-diffusion
> Adding extra search path vae J:\SD\stable-diffusion-webui\models/VAE
> Adding extra search path loras J:\SD\stable-diffusion-webui\models/Lora
> Adding extra search path upscale_models J:\SD\stable-diffusion-webui\models/ESRGAN
> Adding extra search path upscale_models J:\SD\stable-diffusion-webui\models/SwinIR
> Adding extra search path embeddings J:\SD\stable-diffusion-webui\embeddings
> Adding extra search path hypernetworks J:\SD\stable-diffusion-webui\models/hypernetworks
> Adding extra search path controlnet J:\SD\stable-diffusion-webui\extensions\sd-webui-controlnet\models
> Adding extra search path loras J:\SD\stable-diffusion-webui\models\LyCORIS
> Adding extra search path checkpoints R:\SD\Models/Stable-diffusion
> Adding extra search path checkpoints R:\SD\Models/SDXL
> Adding extra search path vae R:\SD\Models/SDXL_VAE
> Adding extra search path loras R:\SD\Models/SDXL_Lora
> Adding extra search path controlnet R:\SD\Models/SDXL_ControlNet

**Comment by pydn** [comment]
Thanks for submitting this! I think this is a great addition. Give me a couple days to review, but I think this should definitely be added to the repo.

File: .gitignore [hunk 1]
```diff
@@ -145,3 +145,6 @@ cython_debug/
 *.sql
 *.sqlite
 *.xml
+/workflow_api.json
+/workflow_api.py
+/custom_nodes/
```
Comment (pydn) [inline .gitignore:150]: I don't think these are needed in .gitignore. Best practice is to just push one file at a time for now.

File: comfyui_to_python.py [hunk 2]
```diff
@@ -10,9 +10,10 @@
 
 import black
 
-from utils import import_custom_nodes, add_comfyui_directory_to_sys_path, get_value_at_index
-
 sys.path.append('../')
+sys.path.append('.')
```
Comment (pydn) [inline comfyui_to_python.py:14]: I think this is not needed. The current working directory will already be added to path.

File: comfyui_to_python.py [hunk 3]
```diff
@@ -306,11 +307,11 @@ def assemble_python_code(self, import_statements: set, speical_functions_code: L
...
         # Get the source code of the utils functions as a string
         func_strings = []
-        for func in [add_comfyui_directory_to_sys_path, get_value_at_index]:
+        for func in [add_comfyui_directory_to_sys_path, get_value_at_index, add_extra_model_paths]:
             func_strings.append(f'\n{inspect.getsource(func)}')
         # Define static import statements required for the script
-        static_imports = ['import os', 'import random', 'import sys', 'from typing import Sequence, Mapping, Any, Union', 
-                          'import torch'] + func_strings + ['\n\nadd_comfyui_directory_to_sys_path()']
+        static_imports = ['import os', 'import random', 'from pathlib import Path', 'import sys', 'from typing import Sequence, Mapping, Any, Union', 
+                          'import torch'] + func_strings + ['\n\nadd_comfyui_directory_to_sys_path()\nadd_extra_model_paths()\n']
```
Comment (pydn) [inline comfyui_to_python.py:314]: Great job cleanly integrating your changes here!

File: utils.py [hunk 4]
```diff
@@ -54,6 +55,39 @@ def search_directory(path: str) -> None:
     # Start the search from the current working directory
     search_directory(start_path)
 
+def add_extra_model_paths() -> Path:
+    """
+    Parse the optional extra_model_paths.yaml file and add the parsed paths to the sys.path.
+    """
+    from pathlib import Path
```
Comment (pydn) [inline utils.py:62]: This import is redundant.

File: utils.py [hunk 5]
```diff
@@ -54,6 +55,39 @@ def search_directory(path: str) -> None:
...
+        # Recursively call the function with the parent directory
+        return find_config_file(parent_directory)
+
+    start_path = os.getcwd()  # Get the current working directory
+    file = find_config_file(start_path)
+    
+    if os.path.isfile(file):
+        load_extra_path_config(file)
+    else:
+        print("Could not find the extra_model_paths config file.")
```
Comment (pydn) [inline utils.py:88]: My main hesitation here is that we could probably generalize the recursive search function into a standalone function and use that function in both add_extra_model_paths() and add_comfyui_directory_to_sys_path(). I think that would be better for following DRY principles.

**Review by pydn — CHANGES_REQUESTED** [review]
I think this looks great, but a couple of small changes will help it fit into the repo.

File: utils.py (hunk 5 above)
Comment (dimtoneff) [reply utils.py:88]: yup, I needed something to work in 10 minutes without duplicating my models and didn't want to touch your func, so I just did this one quickly. Re-using with a general search function would do better.

File: comfyui_to_python.py (hunk 2 above)
Comment (dimtoneff) [reply comfyui_to_python.py:14]: yeah I thought so too but without it the generation did not work. Util's fle could not be imported.

File: utils.py (hunk 5 above)
Comment (pydn) [reply utils.py:88]: totally understand. Let me know if you want to clean this up to be merged, or if you want me to take it and run with it.

File: comfyui_to_python.py (hunk 2 above)
Comment (dimtoneff) [reply comfyui_to_python.py:14]: > PS J:\SD\ComfyUI\ComfyUI\ComfyUI-to-Python-Extension> ..\..\python_embeded\python.exe .\comfyui_to_python.py
Traceback (most recent call last):
  File "J:\SD\ComfyUI\ComfyUI\ComfyUI-to-Python-Extension\comfyui_to_python.py", line 16, in <module>
    from utils import import_custom_nodes, find_path, add_comfyui_directory_to_sys_path, get_value_at_index, add_extra_model_paths
ModuleNotFoundError: No module named 'utils'
PS J:\SD\ComfyUI\ComfyUI\ComfyUI-to-Python-Extension>

**Comment by dimtoneff** [comment]
Hey, it's me again. I cleaned up my code a bit. Please, have a look and change it to fit the repo style if needed, I don't mind (i'm a Python noob.. lol). New code is working great for me now on embedded python Comfy version.

**Comment by pydn** [comment]
Amazing. Thanks for helping out! I'll review and get it merged soon.

**Pull request merged**